src/Microsoft.DotNet.Interactive/Utility/Dotnet.cs

```diff
@@ -1,34 +1,57 @@
 // Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using FluentAssertions.Execution;
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
 using System.Reflection;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.DotNet.Interactive.Utility;
 
 public class Dotnet
 {
     protected readonly DirectoryInfo _workingDirectory;
+    static public string _sdkVersion = null;
 
     public Dotnet(DirectoryInfo workingDirectory = null)
     {
         _workingDirectory = workingDirectory ??
                             new DirectoryInfo(Directory.GetCurrentDirectory());
     }
 
-    public Task<CommandLineResult> New(string templateName, string args = null)
+    public async Task<CommandLineResult> New(string templateName, string args = null)
     {
         if (string.IsNullOrWhiteSpace(templateName))
         {
             throw new ArgumentException("Value cannot be null or whitespace.", nameof(templateName));
         }
 
-        return Execute($@"new ""{templateName}"" {args}");
+        if (_sdkVersion is null)
```

Comment:
sdk version should be driven by the package, not the dotnet class.

Reply:
I'll move the code that creates Global.json to the abstract class Package.

Reply:
done in last commit.